webapp/src/index.tsx

```diff
@@ -79,11 +82,35 @@ export default class Plugin {
         });
     }
 
+    userActivityWatch(): void {
+        // Listen for new activity to trigger a call to the server
+        // Hat tip to the Github and Playbooks plugin
+        let lastActivityTime = Number.MAX_SAFE_INTEGER;
+        const activityTimeout = 60 * 60 * 1000; // 1 hour
+
+        this.activityFunc = () => {
+            const now = new Date().getTime();
+            if (now - lastActivityTime > activityTimeout) {
+                Client.notifyConnect();
+            }
+            lastActivityTime = now;
+        };
+        document.addEventListener('click', this.activityFunc);
+    }
+
     uninitialize() {
         this.removeStoreSubscription?.();
     }
 }
 
+const RootEffects = () => {
+    useEffect(() => {
+        Client.notifyConnect();
+    }, []);
+
```

Comment:
I wonder: given we do this on the first click, do we need this at all? I'm thinking of a scenario where a user's desktop app updates or some such and reloads without their being involved. Should we wait until the user is really around?

Reply:
That's a fair point; I don't see any issue with it—though I'd have to change the click handler logic to be permitted during the first hour.